### ISSUE TITLE
Nightly version ignores allow(unused)

### ISSUE THREAD

**Issue by jo-so** [rust-lang/rust#43910]
The current nightly version (rustc 1.21.0-nightly (df511d554 2017-08-14)) warns about an unused variable, even if it's flagged with allow(). The stable version doesn't do so.
```
fn main() {
    #[allow(unused_variables)] let x = 12;
    let y = 11;
}
```

```
warning: unused variable: `x`
 --> src/main.rs:2:36
  |
2 |     #[allow(unused_variables)] let x = 12;
  |                                    ^
  |
  = note: #[warn(unused_variables)] on by default
```

https://play.rust-lang.org/?gist=9b81da266dd1f4383760de35d94f7ff6&version=nightly

**Comment by Mark-Simulacrum** [comment]
cc @alexcrichton -- would this have been fixed by https://github.com/rust-lang/rust/pull/43841? Not sure if that's present in this nightly.

**Comment by alexcrichton** [comment]
Hm no this was not fixed by #43841, I don't know offhand what might cause this, would have to investigate.

**Comment by nikomatsakis** [comment]
triage: P-high

@alexcrichton, took the liberty of assigning you since it seems likely related to the changes to lint handling.
